src/FlipCounter.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from "react";
+import "./FlipCounter.css";
+class Counter extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      count0: 0,
+      count1: 0,
+      count2: 0,
+      count3: 0,
+      count4: 0,
+      count5: 0,
+      count6: 0
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+  add = () => {
+    if (this.state.count0 == 9) {
+      this.setState({ count0: (this.state.count0 = -1) });
+      this.setState({ count1: this.state.count1 + 1 });
+    }
+
+    if (this.state.count1 == 9) {
+      this.setState({ count1: (this.state.count1 = -1) });
+      this.setState({ count2: this.state.count2 + 1 });
+    }
+
+    if (this.state.count2 == 9) {
+      this.setState({ count2: (this.state.count2 = -1) });
+      this.setState({ count3: this.state.count3 + 1 });
+    }
+
+    if (this.state.count3 == 9) {
+      this.setState({ count3: (this.state.count3 = -1) });
+      this.setState({ count4: this.state.count4 + 1 });
+    }
+
+    if (this.state.count4 == 9) {
+      this.setState({ count4: (this.state.count4 = -1) });
+      this.setState({ count5: this.state.count5 + 1 });
+    }
+
+    this.setState({
+      count0: this.state.count0 + 1
+    });
+  };
+
+  handleChange(event) {
+    this.setState({ value: event.target.value });
+  }
+
+  handleSubmit(event) {
+    this.setState({ count0: parseInt(this.state.value % 10) });
```

Comment:
This is a really bad way to do this task, 'cause it required a lot of code, which will be hard to maintain in the future

The problem is, that you try to denormalize the input just in the beginning and store it denormalized. This is a problem because:

- If you will debug in the future, it is harder to understand what the number in the current state it
- Several `setState` inside one call is a dangerous anti-pattern. Yep, React is clever enough to accumulate these `setState`s when they are inside event handler (like here), but this code is potentially dangerous
- It is hard to understand which digit is first and which digit is last
- It required a lot of code, most of which is duplicated
- The amount of data in RAM is 6 times bigger, than it can be by storing the input data normalised.

What can I advise you to do:

- Store the input value as one number inside the component's state
- Create a function `(input: number) => string[]`, which will receive the input as a number, add leading zeros and return it as an array with 6 strings (one string is one digit of a number)
- Inside render function map this array to render 6 `<td>` elements